Formula/edge-impulse-cli.rb

```diff
@@ -0,0 +1,25 @@
+require "language/node"
+
+class EdgeImpulseCli < Formula
+  desc "Command-line interface tools for Edge Impulse"
+  homepage "https://github.com/edgeimpulse/edge-impulse-cli#readme"
+  url "https://registry.npmjs.org/edge-impulse-cli/-/edge-impulse-cli-1.14.2.tgz"
+  sha256 "ecbd5eb1b5acad313db98cf326349bab6a13f663146f3726bce0f497c30991b8"
+  license "Apache-2.0"
+
+  livecheck do
+    url :stable
+  end
+
+  depends_on "python" => :build
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "1.14.2\n", pipe_output("edge-impulse-blocks --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thank you @SMillerDev since the CLI is interactive and requires credentials, I will do my best to update the tests in a more meaningful way... I will perhaps try to upload files using fake credentials and just check for the auth error coming back.